### PR TITLE
[API-1513] Update the Compact serialization docs with the recent changes

### PR DESCRIPTION
**Pull request by mdumandag**
There were couple of API updates to the Compact serialization,
and it is no longer in BETA status.

This PR updates the documents we have with the recent changes.

### CODE REVIEW

File: docs/modules/serialization/pages/compact-serialization.adoc [hunk 1]
```diff
@@ -266,18 +251,23 @@ public Employee read(CompactReader reader) {
 ----
 
 Then, when faced with binary data serialized by the old writer, new readers will be able to
-read the following fields. Also, Hazelcast provides convenient APIs to read default values
+read the following fields. Also, Hazelcast provides convenient APIs to check  existence of fields in the data.
```
Comment (Serdaro) [inline docs/modules/serialization/pages/compact-serialization.adoc:254]: ... APIs to check the existence ...

File: docs/modules/serialization/pages/compact-serialization.adoc [hunk 2]
```diff
@@ -287,57 +277,17 @@ written by a new writer, they will be able to read all fields.
...
 
-In case of an explicit serializer, you have to supply a type name for the class.
+In case of an explicit serializer, you have to supply a type name for the class
+in the serializer.
 
-Choosing a type name will associate that name with the schema and will make the polyglot
-use cases where there are multiple clients from different languages easier.
+Choosing a type name will associate that name with the schema and will make the
+polyglot use cases where there are multiple clients from different languages
+easier.
```
Comment (Serdaro) [inline docs/modules/serialization/pages/compact-serialization.adoc:290]: .. use cases, where there are .... languages, easier.

**Review by Serdaro — APPROVED** [review]
LGTM, very minor comments.

File: docs/modules/serialization/pages/compact-serialization.adoc [hunk 3]
```diff
@@ -60,13 +51,13 @@ extraction fails, Hazelcast throws an exception.
...
-* Primitive types: `boolean`, `byte`, `short`, `int`, `long`, `float`, and `double`
-* Wrapper classes of primitive types: `Boolean`, `Byte`, `Short`, `Integer`, `Long`, `Float`, and `Double`
+* Primitive types: `boolean`, `byte`, `char`, `short`, `int`, `long`, `float`, and `double`
+* Wrapper classes of primitive types: `Boolean`, `Byte`, `Character`, `Short`, `Integer`, `Long`, `Float`, and `Double`
 * `String`
 * `java.time.LocalDate`, `java.time.LocalTime`, `java.time.LocalDateTime`, and `java.time.OffsetDateTime`
 * `java.math.BigDecimal`
 * Enums
-* Arrays of the types shown above
+* Array, `List`, `ArrayList`, `Set`, `HashSet`, `Map`, `HashMap` of the types shown above
```
Comment (srknzl) [inline docs/modules/serialization/pages/compact-serialization.adoc:60]: Maybe we can add a note about List is serialized/deserialized as ArrayList and so on. If users want another implementation they need to write custom serializer.

File: docs/modules/serialization/pages/compact-serialization.adoc (hunk 1 above)
Comment (srknzl) [inline docs/modules/serialization/pages/compact-serialization.adoc:254]:
```suggestion
read the following fields. Also, Hazelcast provides convenient APIs to check existence of fields in the data.
```

 I don't know excess whitespace is shown in the rendered view but whatever.

**Comment by srknzl** [comment]
Can you remove the `and the Compact Serialization is enabled` parts in https://github.com/hazelcast/hz-docs/blob/50860ca066cd4fde9d4d5d4f55fee386573c967e/docs/modules/serialization/pages/serialization.adoc#how-hazelcast-serializes-objects

**Review by srknzl — CHANGES_REQUESTED** [review]

File: docs/modules/serialization/pages/compact-serialization.adoc (hunk 3 above)
Comment (mdumandag) [reply docs/modules/serialization/pages/compact-serialization.adoc:60]: Yes, I will do that, along with lots of other documentation updates with Compact before the GA. This PR is there to make sure the code snippets/docs we have are up to date

**Comment by mdumandag** [comment]
@srknzl I have addressed your comments, can you re-review again?

**Review by srknzl — APPROVED** [review]

**Pull request merged**